### ISSUE TITLE
stderr archiveLogs not persisting on artifact

### ISSUE THREAD

**Issue by akshaybhatt14495** [argoproj/argo#1775]
**Is this a BUG REPORT or FEATURE REQUEST?**:
BUG REPERT

**What happened**:
I enabled log archiving via the workflow-controller-configmap ConfigMap with:
```
apiVersion: v1
kind: ConfigMap
metadata:
  name: workflow-controller-configmap
  namespace: argo
data:
  config: |
    artifactRepository:
      archiveLogs: true
      s3:
        bucket: argo-poc-logs
        endpoint: s3.endpoint
        accessKeySecret:
          name: s3-credentials
          key: access-key
        secretKeySecret:
          name: s3-credentials
          key: secret-key
```
**What you expected to happen**:
Full logs written to s3. (stdout + stderr)

**How to reproduce it (as minimally and precisely as possible)**:
1: Enable archive logging as shown in ConfigMap above
2: Submit a workflow with 
     archivelocation: 
          archivelogs : true
3: Once workflow completes, delete pod(s)

**Anything else we need to know?**:

**Environment**:
- Argo version:
```
$ argo version
argo: v2.4.2
  BuildDate: 2019-10-21T18:39:52Z
  GitCommit: 675c66267f0c916de0f233d8101aa0646acb46d4
  GitTreeState: clean
  GitTag: v2.4.2
  GoVersion: go1.11.5
  Compiler: gc
  Platform: darwin/amd64
```
- Kubernetes version :
```
$ kubectl version -o yaml
clientVersion:
  buildDate: "2019-10-15T23:41:55Z"
  compiler: gc
  gitCommit: c97fe5036ef3df2967d086711e6c0c405941e14b
  gitTreeState: clean
  gitVersion: v1.16.2
  goVersion: go1.12.10
  major: "1"
  minor: "16"
  platform: darwin/amd64
serverVersion:
  buildDate: "2019-06-19T16:32:14Z"
  compiler: gc
  gitCommit: e8462b5b5dc2584fdcd18e6bcfe9f1e4d970a529
  gitTreeState: clean
  gitVersion: v1.15.0
  goVersion: go1.12.5
  major: "1"
  minor: "15"
  platform: linux/amd64
```

**Comment by tcolgate** [comment]
may be fixed by #2136

**Comment by alexec** [comment]
Can you see if this is still an issue in v2.5? Is so - please re-open.
